### ISSUE TITLE
zenmode: hiding header

### ISSUE THREAD

**Issue by lsloan** [ipython-contrib/jupyter_contrib_nbextensions#1309]
When the "Hide the header in zenmode" option is enabled (which it is by default) but the notebook's header is **_already hidden_** before zenmode is entered, then the headers will be **_shown_** when zenmode exits.  Additionally, the "View > Toggle Header" menu item must then be selected **_twice_** to hide the header again.

I think zenmode needs to check the visibility of the header when it is entered and save that state for later use.  If the header is already hidden, zenmode shouldn't try to hide it.  Likewise, later when zenmode exits, if the header was initially hidden, zenmode shouldn't try to show it.

**Comment by jcb91** [comment]
True, zenmode was written before the header could be hidden by any other means, but now it needs to check!
